src/vs/workbench/api/common/extHostStatusBar.ts

```diff
@@ -272,6 +288,10 @@ export class ExtHostStatusBarEntry implements vscode.StatusBarItem {
 
 	public dispose(): void {
 		this.hide();
+
+		this.staticItems.delete(this._entryId);
+		this.entries.delete(this._entryId);
```

Comment:
This feels a bit off: you are passing `entries` in and even keep it as `private` (which means extensions could access it) to remove from the `dispose`. This needs a different, safer approach so that extensions cannot tamper.

Reply:
I am leveraging the `$acceptStaticEntries()` to cleanup the items from the `entries` map.